docs/canvas/canvas-expression-lifecycle.asciidoc

```diff
@@ -0,0 +1,259 @@
+[role="xpack"]
+[[canvas-expression-lifecycle]]
+== Canvas expression lifecycle
+
+Elements in Canvas are all created using an *expression language* that defines how to retrieve, manipulate, and ultimately visualize data. The goal is to allow you to do most of what you need without understanding the *expression language*, but learning how it works unlocks a lot of Canvas's power.
+
+
+[[canvas-expressions-always-start-with-a-function]]
+=== Expressions always start with a function
+
+Expressions simply execute <<canvas-function-reference, functions>> in a specific order, which produce some kind of output. That output can then be inserted into another function, and another after that, until it produces the output you need. 
+
+To use random available data in Canvas to produce a table, run the following expression:
+
+[source,text]
+----
+filters
+| demodata
+| table
+| render
+----
+
+This expression starts out with the <<filters_fn, filters>> function, which provides the value of any time filters or dropdown filters in the workpad. This is then inserted into <<demodata_fn, demodata>>, a function that returns exactly what you expect, demo data. Because the <<demodata_fn, demodata>> function receives the filter information from the <<filters_fn, filters>> function before it, it applies those filters to reduce the set of data it returns. We call the output from the previous function _context_.
```

Comment:
```suggestion
This expression starts out with the <<filters_fn, filters>> function, which provides the value of any time filters or dropdown filters in the workpad. This is then piped into <<demodata_fn, demodata>>, a function that returns exactly what you expect, demo data. Because the <<demodata_fn, demodata>> function receives the filter information from the <<filters_fn, filters>> function before it, it applies those filters to reduce the set of data it returns. We call the output from the previous function _context_.
```

> piped

We do call the output of previous function _context_ currently, but we were discussing with @ppisljar that maybe its better to call it _input_, as usually context refers to something that would be shared across all functions, and input would be something that a function receives, like STDIN in command line tools. I'm just adding a mental note here for myself, not suggesting to do anything regarding this PR.